pkg/apis/autoscaling/v1/conversion.go

```diff
@@ -375,14 +390,17 @@ func Convert_v1_HorizontalPodAutoscaler_To_autoscaling_HorizontalPodAutoscaler(i
 
 		// the normal Spec conversion could have populated out.Spec.Metrics with a single element, so deal with that
 		outMetrics := make([]autoscaling.MetricSpec, len(otherMetrics)+len(out.Spec.Metrics))
+		if out.Spec.Metrics != nil && cpuMetricIndex < len(outMetrics) {
+			outMetrics[cpuMetricIndex] = out.Spec.Metrics[0]
+		}
 		for i, metric := range otherMetrics {
+			if out.Spec.Metrics != nil && i >= cpuMetricIndex {
+				i = i + 1
```

Comment:
Please don't increment for loop variables inside the loop.

Reply:
OK, I remove this out of  `for loop`. Thanks for your advice.